course/steps/03-clients-analytics/en.md

```diff
@@ -0,0 +1,107 @@
+
+It's possible to read more about clients concepts [on this document](https://www.notion.so/How-to-use-and-create-Clients-on-VTEX-IO-3598e97a761645e0befdac84a32f339d).
+
+## Analytics client
+
+In this course, it will be necessary to create a client that will be used to get information regardning product's number of views. The client that will be created will make a REST request in which it'll retrieve information about product views. This client needs to have a handler that will be response to handle the request made on a specific route and this is how it can be tested.
+
+## Activity
+
+In this step, we will implement the Anaylitcs client.So,
+
+1. First, in the `/node/clients/` directory, you will find a file called `analytics.ts`, which already has an sketch, just like the code block below. This is where you'll implement your client.
+
+   ```ts
+   import { AppClient } from '@vtex/api'
+
+   export default class Analytics extends AppClient {}
+   ```
+
+   > You can noticed in this code block that Analytics is a client that extends from `AppClient`.
+
+2. The client needs to have a constructor and just a single method, called `getLiveUsers`. This method returns a promise of an array that its elements are of the type `LiveUsersProduct`. Using the code below, add the necessary code lines to the client:
+
+   ```diff
+   //node/clients/analytics.ts
+   import { AppClient, InstanceOptions, IOContext } from '@vtex/api'
+
+   export default class Analytics extends AppClient {
+   +  constructor(context: IOContext, options?: InstanceOptions) {
+   +    super('vtex.mocked-analytics@0.x', context, options)
+   +  }
+
+   +  public getLiveUsers(): Promise<LiveUsersProduct[]> {}
+   }
+
+   +interface LiveUsersProduct {
+   +  slug: string
+   +  liveUsers: number
+   +}
+   ```
+
+   > The interface that is defined is going to be used as a typing on the method that we'll implement.
+
+3. Now it's necessary to implement the `getLiveUsers` method. It **returns** a HTTP GET request to a well-defined endpoint that is responsible for getting the data that is needed in this application. So add the following line to the method `getLiveUsers`:
+
+   ```ts
+   return this.http.get('_v/live-products')
+   ```
+
+   > The method that you've just created is will get the necessary data for this application: an array of objects that have two fields: `slug`, a string that represents the product ID and `liveUsers`, a number that is the quantity of users visualizing this product - which are the fields in the interface.
+
+4. With your analytics client already implemented, it's necessary to declare it as one of the clients in the `Clients` class, so it will be accessible by the `Context` that we talked about at the beginning of this step.
```

Comment:
```suggestion
4. With your analytics client already implemented, it's necessary to declare it as one of the clients in the `Clients` class, so it will be accessible using the `Context` that we've talked about at the beginning of this step.
```